pkg/client/client_v2_test.go

```diff
@@ -209,51 +187,48 @@ func TestClientV2_Checkin_Initial(t *testing.T) {
 					units = append(units, change.Unit)
 					unitsMu.Unlock()
 				default:
-					panic("not implemented")
+					panic(fmt.Errorf("received unnexpected change type: %s",
+						change.Type))
 				}
 			}
 		}
 	}()
 
-	require.NoError(t, validClient.Start(ctx))
-	defer validClient.Stop()
-	require.NoError(t, waitFor(func() error {
-		m.Lock()
-		defer m.Unlock()
-		unitsMu.Lock()
-		defer unitsMu.Unlock()
-
-		if !gotValid {
-			return fmt.Errorf("server never received valid token")
-		}
-		if len(units) < 2 {
-			return fmt.Errorf("never received 2 units")
-		}
-		return nil
-	}))
-
-	agentInfo := validClient.AgentInfo()
-	require.NotNil(t, agentInfo)
+	err := clientv2.Start(ctx)
+	require.NoError(t, err, "could not start client")
+	defer clientv2.Stop()
+
+	var gotObserved *proto.CheckinObserved
+	timeout := 1 * time.Second
+	select {
+	case <-time.After(timeout):
+		t.Fatalf("timedou out after %s waiting for 1st checkin to complete",
+			timeout)
+	case gotObserved = <-checkInDone:
+	}
 
-	validClient.Stop()
+	agentInfo := clientv2.AgentInfo()
+	clientv2.Stop()
 	cancel()
 
-	assert.Equal(t, agentInfo.ID, "elastic-agent-id")
-	assert.Equal(t, agentInfo.Version, "8.5.0")
-	assert.True(t, agentInfo.Snapshot)
+	assert.Empty(t, err, "client returned errors")
 
-	assert.Equal(t, wantFQDN, gotFQDN)
+	require.NotNil(t, agentInfo,
+		"agent info is nil, it should have been set during check-in")
+	assert.Equal(t, agentInfo.Version, packageVersion)
+	assert.True(t, agentInfo.Snapshot, "agent version is a SNAPSHOT")
+
+	assert.True(t, gotFQDN, "FQDN should be enabled")
```

Comment:
There should probably be an assert.Len(t, units, 2) right after this since the block above that did this was removed. The test would fail without this, just with an index error below.

```go
if len(units) < 2 {
     return fmt.Errorf("never received 2 units")
}
```